src/dwarf2pdb.cpp

```diff
@@ -876,39 +946,120 @@ int CV2PDB::addDWARFStructure(DWARF_InfoData& structid, DWARF_CompilationUnit* c
 	return cvtype;
 }
 
-int CV2PDB::getDWARFArrayBounds(DWARF_InfoData& arrayid, DWARF_CompilationUnit* cu,
-								DIECursor cursor, int& upperBound)
+void CV2PDB::getDWARFArrayBounds(DWARF_InfoData& arrayid, DWARF_CompilationUnit* cu,
+								DIECursor cursor, int& basetype, int& lowerBound, int& upperBound)
 {
-	int lowerBound = 0;
+	DWARF_InfoData id;
 
+	// TODO: handle multi-dimensional arrays
 	if (cu)
 	{
-		DWARF_InfoData id;
 		while (cursor.readNext(id, true))
 		{
-			int cvid = -1;
 			if (id.tag == DW_TAG_subrange_type)
 			{
-				lowerBound = id.lower_bound;
-				upperBound = id.upper_bound;
+				getDWARFSubrangeInfo(id, cu, basetype, lowerBound, upperBound);
+				return;
 			}
 			cursor.gotoSibling();
 		}
 	}
-	return lowerBound;
+
+	// In case of error, return plausible defaults
+	getDWARFSubrangeInfo(id, NULL, basetype, lowerBound, upperBound);
+}
+
+void CV2PDB::getDWARFSubrangeInfo(DWARF_InfoData& subrangeid, DWARF_CompilationUnit* cu,
+	int& basetype, int& lowerBound, int& upperBound)
+{
+	// In case of error, return plausible defaults
+	basetype = T_INT4;
+	lowerBound = currentDefaultLowerBound;
+	upperBound = 0;

```

Comment:
Should `upperBound` default to the same value as `lowerBound` or `lowerBound-1`?

Reply:
Good question: I’m not sure what makes most sense. I’ll go for `lowerBound-1` so that by default we get an empty array, but maybe an array with one element would be useful for users facing bogous debug info… What do you think?

Reply:
If it is a case of bogus debug information anyway, showing the first element of the array could be more helpful. So I'd propose `upperBound = lowerBound`.

Reply:
Thank you! I’ll update this part accordingly.